src/Raven.Server/Documents/TcpHandlers/SubscriptionConnection.cs

```diff
@@ -516,6 +575,10 @@ private async Task<SubscriptionConnectionClientMessage> GetReplyFromClientAsync(
                     return JsonDeserializationServer.SubscriptionConnectionClientMessage(blittable);
                 }
             }
+            catch (EndOfStreamException e)
```

Comment:
please check if it doesn't break anything. quite risky for me.

Reply:
Yeah, what was the reasoning for catching it explicitly? I wonder if that cannot break error handling on the client side

Reply:
This catch was added for the following scenario:

When the client worker app (the subscriber that processes the docs received) 
throws an exception (for whatever reason)
it comes down to an `EndOfStreamException `that is thrown on the server-side
` "Stream ended w/o reaching end of json content"`
which is thrown from method `ParseToMemoryAsync` in `jsonOperationContext.cs`

It doesn't seem to break anything since the exception is caught anyway and handled correctly with method
`ReportExceptionToClient`
 
The only difference is that I can expose in UI an error that has more meaning 
("No reply from the subscription client" instead of the unclear "end of json content" msg)

Reply:
Okay. So this error isn't sent to the subscription client - because the connection is already down. We might be better off by creating a dedicated exception for that - for example `SubscriptionConnectionDownException` - see abstract `SubscriptionException` and its implementations.

Reply:
done - added `SubscriptionConnectionDownException `